# HG changeset patch
# User Soren Hauberg  <hauberg@gmail.com>
# Date 1231997778 18000
#      Thu Jan 15 00:36:18 2009 -0500
# Node ID 715f6cd8d5e7d51de0cc22ade86e89552ed490cd
# Parent  68aa5abfd13640f40b5ca5f13321dfe458ff1051
__magick_read__.cc (encode_uint_image): initialize bitdepth

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2009-01-15  Søren Hauberg  <hauberg@gmail.com>
+
+	* DLD-FUNCTIONS/__magick_read__.cc (encode_uint_image):
+	Initialize bitdepth.
+
 2009-01-14  Søren Hauberg  <hauberg@gmail.com>
 
 	* DLD-FUNCTIONS/betainc.cc, DLD-FUNCTIONS/chol.cc,
 	DLD-FUNCTIONS/daspk.cc, DLD-FUNCTIONS/dasrt.cc,
 	DLD-FUNCTIONS/dassl.cc, DLD-FUNCTIONS/filter.cc,
 	DLD-FUNCTIONS/gammainc.cc, DLD-FUNCTIONS/gcd.cc,
 	DLD-FUNCTIONS/givens.cc, DLD-FUNCTIONS/hess.cc,
 	DLD-FUNCTIONS/lsode.cc, DLD-FUNCTIONS/qr.cc, DLD-FUNCTIONS/qz.cc,
diff --git a/src/DLD-FUNCTIONS/__magick_read__.cc b/src/DLD-FUNCTIONS/__magick_read__.cc
--- a/src/DLD-FUNCTIONS/__magick_read__.cc
+++ b/src/DLD-FUNCTIONS/__magick_read__.cc
@@ -536,17 +536,17 @@ encode_bool_image (std::vector<Magick::I
 }
 
 template <class T>
 static void
 encode_uint_image (std::vector<Magick::Image>& imvec,
                    const octave_value& img,
                    bool has_map)
 {
-  unsigned int bitdepth;
+  unsigned int bitdepth = 0;
   T m;
 
   if (img.is_uint8_type ())
     {
       bitdepth = 8;
       m = img.uint8_array_value ();
     }
   else if (img.is_uint16_type ())
